crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -24,6 +24,18 @@ impl Direction2d {
         Self::new_and_length(value).map(|(dir, _)| dir)
     }
 
+    /// Create a [`Direction2d`] from a [`Vec2`] that is already normalized.
+    ///
+    /// ## Safety
+    ///
+    /// This function only checks if `value` is normalized on debug builds,
+    /// release builds can allow the creation of a [`Direction2d`] that is not normalized.
+    pub unsafe fn new_unchecked(value: Vec2) -> Self {
```

Comment:
This presents nothing that is actually undefined behavior in Rust. We should not be making APIs like these.